### PR TITLE
Changed so that (only) lua 5.2.x gets accepted

### PR DESCRIPTION
**Pull request by beedaddy**
On Arch Linux lua 5.2.4 has not been detected. So I removed the EXACT argument and added a check so that lua 5.3.x get rejected (as it seems to be intended by the devs).

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -456,8 +456,8 @@ include_directories("${CMAKE_CURRENT_BINARY_DIR}" "${CMAKE_CURRENT_SOURCE_DIR}")
 # Build external deps
 #
 if(USE_LUA)
-  find_package(Lua52 5.2 EXACT)
-  if(LUA52_FOUND)
+  find_package(Lua52 5.2)
+  if(LUA52_FOUND AND LUA52_VERSION LESS 5.3)
```
Comment (LebedevRI) [inline src/CMakeLists.txt]: if HAND is found and HAND is not a FOOT :)

**Comment by beedaddy** [comment]
@LebedevRI Does it look better now? :)

**Comment by LebedevRI** [comment]
@beedaddy well, yes, certainly looks better to me, BUT.
Now i do not like that we pass some version, and then use a magical number (5.3).
Maybe it is better to pass 5.3 too, or compute it from 5.2...

@boucman what do you think?

Though, maybe i'm just too picky :)

**Comment by boucman** [comment]
@LebedevRI I'll have to disagree on that one... 5.3 is not really a magical number, it's the first version that is not 5.2 

With lua we usually want an exact major version since compatibility breaks, and in the case of 5.2 we know for a fact that it does, since 5.3 is released. To me it makes little sense to specify it at the call level since it won't ever be any value except 5.3

It could be argued that we should be able to test either =5.2 or >=5.2, but this is a DT implementation of FindLua5.2 and in our case we want exactly 5.2 until we migrate and want exactly 5.3

And thx for reviving this PR, I had forgotten about it :)

**Pull request merged**